electrum/lnworker.py

```diff
@@ -1238,11 +1236,11 @@ async def pay_to_node(
                 #       instead we should give feedback to create_routes_for_payment.
                 if code in (OnionFailureCode.TRAMPOLINE_FEE_INSUFFICIENT,
                             OnionFailureCode.TRAMPOLINE_EXPIRY_TOO_SOON):
-                    # todo: parse the node parameters here (not returned by eclair yet)
-                    trampoline_fee_levels[erring_node_id] += 1
+                    # TODO: parse the node policy here (not returned by eclair yet)
+                    # TODO: erring node is always the first trampoline even if second
+                    #  trampoline demands more fees, we can't influence this
+                    trampoline_fee_level += 1
                     continue
-                elif use_two_trampolines:
-                    use_two_trampolines = False
```

Comment:
I am trying to understand how exactly this fallback from two to one TFs works atm, and whether it is related to https://github.com/spesmilo/electrum/issues/7669.
AFAIU the logic here, if the failures were trampoline-feelevel-related, that is not cause in itself to fallback to one TF, but we would have fallen back if the first TF cannot find a route to the second one or if the second one cannot find a route to the destination, right? One thing to consider is given we only know of 3 TFs atm, it's plausible that some of the time  there aren't two TFs online. Although maybe the current code is falling back too soon -- say TF_Alice is offline and we tried to route through it, instead of immediately falling back to using one TF, we could still try selecting a new route of two TFs that excludes Alice.
Anyway, what is the motivation for removing this?

Reply:
I forgot to add it back. Right, we could do better than immediately falling back to a single TF, in principle we could track whether a TF was able to deliver a certain amount to a specific target and then remove it from the list of possible second forwarders.

Reply:
The fallback in the legacy case is added back now, hope it looks better now.